### PR TITLE
feat: add capture.py - also fixes audio recording

### LINKED ISSUE #183: Error getting focused window (macOS)
I noticed that record & replay fail when in Google Meets, but this also extends to calls like Discord (with no video being sent)

To reproduce:  I just got into an empty discord voice channel and ran NaiveReplay. This still occurs if I am "deafened" (no microphone or audio being transmitted)

An example of the message, followed by a segfault:
```bat
2023-05-27 01:36:14.192 | ERROR    | openadapt.window._macos:get_active_window:69 - Error getting focused window
zsh: segmentation fault  python3 -m openadapt.record "testing"
```
However, the recording still proceeds and stalls at this stage:
 ```bat
 2023-05-27 01:36:15.198 | INFO     | __mp_main__:write_events:215 - event_type='window' starting
2023-05-27 01:36:15.221 | INFO     | __mp_main__:write_events:215 - event_type='screen' starting
2023-05-27 01:36:15.228 | INFO     | __mp_main__:write_events:215 - event_type='action' starting
 ```
 I noticed my fans started spinning more aggressively after failed recordings, and received this from `ps`
 ```bat
   PID TTY           TIME CMD
 2591 ttys000    0:00.13 -zsh
 2651 ttys000    0:00.03 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.resource_tracker import main;main(12)
 2652 ttys000   12:16.87 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 2653 ttys000   12:21.22 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 2654 ttys000   12:21.63 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 2682 ttys000    0:00.03 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.resource_tracker import main;main(9)
 2683 ttys000   12:14.65 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 2684 ttys000   12:14.18 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 2685 ttys000   12:14.90 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 3279 ttys000    0:00.03 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.resource_tracker import main;main(9)
 3280 ttys000    1:40.91 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 3281 ttys000    1:40.87 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 3282 ttys000    1:36.87 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 3390 ttys000    0:00.03 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.resource_tracker import main;main(9)
 3391 ttys000    0:03.96 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 3392 ttys000    0:04.33 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 3393 ttys000    0:04.17 /opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/Resources/Python.app/Contents/MacOS/Python -c from multiprocessing.spawn import spawn_main; spawn_main(tra
 ```
 Uh oh
 should  `pkill -f python`
result:

```bat
/opt/homebrew/Cellar/python@3.10/3.10.11/Frameworks/Python.framework/Versions/3.10/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 17 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '
``` 
 (x4),
  but processes are all still killed.
 
 Workarounds:
 Just leaving the call instantly resolves all issues, but this isn't the best approach.
 
 Possible fixes:
 - Notify the user that something is wrong, and use a fall-back recording method without window states.
 - Kill the process via terminate() / SIGTERM if we failed to capture window state
 
 Best solution would be to get around this entirely, but I'm not entirely sure what is actually causing this, since other actions like streaming music on spotify has no effect on this.

### PR DESCRIPTION
**Pull request by 0dm**
**What kind of change does this PR introduce?**
Adds a new module - capture for macOS. This allows for easy screen + audio recording. 

```py
    capture = OpenAdaptCapture()
    capture.start(audio=True)
    input("Press enter to stop")
    capture.stop()
```

### bugfix: we can now record audio while also recording window states.

**Summary**

motivation: i dug through apple's api for what felt like ages
creating this allowed me to easily debug and find the cause of the error getting focused window bug - closes #183 
this is also useful if we ever need a recording.

**Checklist**
* [x] My code follows the style guidelines of OpenAdapt
* [x] I have performed a self-review of my code
* [x] If applicable, I have added tests to prove my fix is functional/effective
* [x] I have linted my code locally prior to submission
* [x] I have commented my code, particularly in hard-to-understand areas
* [x] I have made corresponding changes to the documentation (e.g. README.md, requirements.txt)
* [x] New and existing unit tests pass locally with my changes

**How can your code be run and tested?**

demo: https://streamable.com/5a09ym
captured via capture.py
you can clearly hear the audio!

**Other information**

### CODE REVIEW

**Comment by abrichr** [comment]
Than you @0dm! Any thoughts on how to implement this on Windows? Do you have a Windows machine you can use?

**Comment by 0dm** [comment]
@abrichr I can try working on a Windows version (win32) using Parallels.

File: openadapt/replay.py [hunk 1]
```diff
@@ -66,8 +69,26 @@ def replay(
     strategy = strategy_class(recording)
     logger.info(f"{strategy=}")
 
-    strategy.run()
-    return True
+    handler = None
+    rval = True
+    if record:
+        capture.start(audio=False, camera=False)
+        sleep(1)
```
Comment (abrichr) [inline openadapt/replay.py:76]:
```suggestion
        # TODO: handle this more robustly
        sleep(1)
```

File: openadapt/replay.py [hunk 2]
```diff
@@ -66,8 +69,26 @@ def replay(
...
+        handler = logger.add(
+            open(f"captures{sep}log-{strategy_name}-{recording.timestamp}.log", "w")
```
Comment (abrichr) [inline openadapt/replay.py:78]:
```suggestion
            file_name = f"log-{strategy_name}-{recording.timestamp}.log"
            # TODO: make configurable
            dir_name = "captures"
            file_path = os.path.join(dir_name, file_name)
            open(file_path, "w")
```

File: openadapt/replay.py [hunk 3]
```diff
@@ -66,8 +69,31 @@ def replay(
...
+        capture.start(audio=False, camera=False)
+        # TODO: handle this more robustly
+        sleep(1)
+        handler = logger.add(
+            file_name = f"log-{strategy_name}-{recording.timestamp}.log"
+            # TODO: make configurable
+            dir_name = "captures"
+            file_path = os.path.join(dir_name, file_name)
+            open(file_path, "w")
+        )
```
Comment (abrichr) [inline openadapt/replay.py:84]:
```suggestion
        file_name = f"log-{strategy_name}-{recording.timestamp}.log"
        # TODO: make configurable
        dir_name = "captures"
        file_path = os.path.join(dir_name, file_name)
        logger.info(f"{file_path=}")
        handler = logger.add(
            open(file_path, "w")
        )
```

**Review by abrichr — APPROVED** [review]
LGTM! 🚀

**Pull request merged**